pkg/cloud/cloud.go

```diff
@@ -1621,25 +1677,55 @@ func (c *cloud) waitForVolumeModification(ctx context.Context, volumeID string)
 	return nil
 }
 
+func describeVolumesModifications(ctx context.Context, svc EC2API, request *ec2.DescribeVolumesModificationsInput) ([]types.VolumeModification, error) {
+	volumeModifications := []types.VolumeModification{}
+	var nextToken *string
+	for {
+		response, err := svc.DescribeVolumesModifications(ctx, request)
+		if err != nil {
+			if isAWSErrorModificationNotFound(err) {
+				return nil, VolumeNotBeingModified
+			}
+			return nil, fmt.Errorf("error describing volume modifications: %w", err)
+		}
+
+		volumeModifications = append(volumeModifications, response.VolumesModifications...)
+
+		nextToken = response.NextToken
+		if aws.ToString(nextToken) == "" {
+			break
+		}
+		request.NextToken = nextToken
+	}
+	return volumeModifications, nil
+}
+
 // getLatestVolumeModification returns the last modification of the volume.
-func (c *cloud) getLatestVolumeModification(ctx context.Context, volumeID string) (*types.VolumeModification, error) {
+func (c *cloud) getLatestVolumeModification(ctx context.Context, volumeID string, isBatched bool) (*types.VolumeModification, error) {
 	request := &ec2.DescribeVolumesModificationsInput{
 		VolumeIds: []string{volumeID},
 	}
-	mod, err := c.ec2.DescribeVolumesModifications(ctx, request)
-	if err != nil {
-		if isAWSErrorModificationNotFound(err) {
+
+	// TODO Q: I see this as the cleanest way to NOT batch certain DVM calls.
+	// TODO Q Cont: Would making a separate batcher with maxEntries 1 or maxDelay 0 be cleaner?
+	if c.bm == nil || !isBatched {
+		mod, err := c.ec2.DescribeVolumesModifications(ctx, request)
+		if err != nil {
+			if isAWSErrorModificationNotFound(err) {
+				return nil, VolumeNotBeingModified
+			}
+			return nil, fmt.Errorf("error describing modifications in volume %q: %w", volumeID, err)
+		}
+
+		volumeMods := mod.VolumesModifications
+		if len(volumeMods) == 0 {
 			return nil, VolumeNotBeingModified
 		}
-		return nil, fmt.Errorf("error describing modifications in volume %q: %w", volumeID, err)
-	}
 
-	volumeMods := mod.VolumesModifications
-	if len(volumeMods) == 0 {
-		return nil, VolumeNotBeingModified
+		return &volumeMods[len(volumeMods)-1], nil // TODO Q Check for nil pointer shenanigan
```

Comment:
If I understand the question correctly - no need for that check since it is guaranteed that length of volumeMods > 0 at this point.